clients/vault/src/stellar_oracle.rs

```diff
@@ -0,0 +1,507 @@
+#![allow(dead_code)]
+
+use std::{
+    collections::{btree_map::Keys, BTreeMap, HashMap},
+    convert::{TryFrom, TryInto},
+    fs,
+    fs::File,
+    io::{Read, Write},
+    path::PathBuf,
+    str::Split,
+};
+
+use crate::error::Error;
+use stellar_relay::{
+    connect,
+    helper::{compute_non_generic_tx_set_content_hash, time_now},
+    node::NodeInfo, ConnConfig, StellarNodeMessage, UserControls,
+};
+
+use stellar_relay::sdk::{
+    compound_types::UnlimitedVarArray,
+    network::Network,
+    types::{
+        ScpEnvelope, ScpStatementExternalize, ScpStatementPledges, StellarMessage, TransactionSet, Uint64,
+    },
+    Hash, XdrCodec,
+};
+
+pub type Slot = Uint64;
+pub type TxSetHash = Hash;
+pub type TxHash = Hash;
+pub type SerializedData = Vec<u8>;
+pub type Filename = String;
+
+/// For easy writing to file. BTreeMap to preserve order of the slots.
+pub type SlotEncodedMap = BTreeMap<Slot, SerializedData>;
+
+/// The slot is not found in the `StellarMessage::TxSet(...)`, therefore this map
+/// serves as a holder of the slot when we hash the txset.
+pub type TxSetCheckerMap = HashMap<TxSetHash, Slot>;
+
+/// Todo: these maps should be merged into one; but there was a complication (which differs in every run):
+/// Sometimes not enough `StellarMessage::ScpMessage(...)` are sent per slot;
+/// or that the `Stellar:message::TxSet(...)` took too long to arrive (may not even arrive at all)
+/// So I've kept both of them separate.
```

Comment:
I would remove this 'TODO' comment. Yes, it was originally intended to have one map but it is not important, so maybe change it to something like 'There are separate maps for TX sets and SCP envelopes because...'

Reply:
I forgot to reply to this :scream:
I did not want to extend the `get()` function, because the map shouldn't be allowed to access the files. Instead, I have FileHandlers that can be used to search for  transactions not found in the map.


For the tests... it's a todo. :bow: